Homework3: Anton Stychnevsky

Stychnevsky opened the pull request:
Homework # 3

Stychnevsky committed:
Homework # 3

belovmd commented on HW3-task1.py:
```diff
@@ -0,0 +1,20 @@
+is still missing it. """
+
+
+list1 = [lett1 + lett2 for lett1 in ['a', 'b'] for lett2 in ['b', 'c', 'd']]
```
'ab' etc?

belovmd commented on HW3-task1.py:
```diff
@@ -0,0 +1,20 @@
+list1 = [lett1 + lett2 for lett1 in ['a', 'b'] for lett2 in ['b', 'c', 'd']]
+sliced_list = list1[::2]
+
+list2 = [(str(pos) + 'a') for pos in range(1, 5)]
```
we have more native way instead of range

belovmd commented on HW3-task2.py:
```diff
@@ -0,0 +1,22 @@
+
+tuple2 = ('a', 'b', 'c')
+lst2_1 = list(tuple2)  # way1
+lst2_2 = [i for i in tuple2]  # way2, using generators
```
it's a good tip, but usually we just use list constructor

belovmd commented on HW3-task3.py:
```diff
@@ -0,0 +1,22 @@
+
+
+def count_characters(count_me_string):
+    return {lett: count_me_string.count(lett) for lett in set(count_me_string)}
```
it's O(n**2) complexity.
we have O(n) solution

belovmd reviewed (commented):
please address these minor issues

Stychnevsky committed:
Improved according to recommendations

Task1: strings instead of lists and range
Task2: deleted second way to create a list from tuple
Task3: changed function using dictionary

belovmd reviewed (commented):
8.5
it's maximal score for full completion of this hw.
